### PR TITLE
add action with custom key support

### PR DESCRIPTION
**Pull request by Dmetriy-MC**
Added the ability to add actions with a custom key. For example, to call NEI directly from the map.

### CODE REVIEW

File: src/main/java/com/gtnewhorizons/navigator/api/model/layers/UniversalInteractableRenderer.java [hunk 1]
```diff
@@ -104,10 +104,15 @@ public boolean onKeyPressed(int keyCode) {
...
-            return true;
+        if (hoveredRenderStep != null) {
+            if (Util.isKeyPressed(NavigatorApi.ACTION_KEY)) {
+                hoveredRenderStep.onActionKeyPressed();
+                manager.forceRefresh();
+                return true;
+            } else if (hoveredRenderStep.onActionWithKeyPressed(keyCode)) {
+                manager.forceRefresh();
+                return true;
+            }
```
Comment (Lyfts) [inline src/main/java/com/gtnewhorizons/navigator/api/model/layers/UniversalInteractableRenderer.java]: If you follow the other example then all of these can be turned into

```suggestion
       if (hoveredRenderStep != null && hoveredRenderStep.onKeyPressed(keyCode)) {
            manager.forceRefresh();
            return true;
        }
```

File: src/main/java/com/gtnewhorizons/navigator/api/model/steps/InteractableStep.java [hunk 2]
```diff
@@ -10,6 +10,8 @@ public interface InteractableStep extends RenderStep {
 
     void onActionKeyPressed();
 
+    boolean onActionWithKeyPressed(int keyCode);
```
Comment (Lyfts) [inline src/main/java/com/gtnewhorizons/navigator/api/model/steps/InteractableStep.java]: This needs to be a default method or all the dependents will break, I also think `onKeyPressed` is a more fitting name for this method.
You can change it to something like the following which means you can also remove all the references to `onActionKeyPressed` from the various LayerRenderers. If you do this then `onActionKeyPressed` should be turned into an empty default method.

```suggestion
       default boolean onKeyPressed(int keyCode) {
        if (Util.isKeyPressed(NavigatorApi.ACTION_KEY)) {
            onActionKeyPressed();
            return true;
        }

        return false;
    }
```

**Review by Lyfts — CHANGES_REQUESTED** [review]

File: src/main/java/com/gtnewhorizons/navigator/api/model/layers/UniversalInteractableRenderer.java (hunk 1 above)
Comment (Dmetriy-MC) [reply src/main/java/com/gtnewhorizons/navigator/api/model/layers/UniversalInteractableRenderer.java]: Fixed

File: src/main/java/com/gtnewhorizons/navigator/api/model/steps/InteractableStep.java (hunk 2 above)
Comment (Dmetriy-MC) [reply src/main/java/com/gtnewhorizons/navigator/api/model/steps/InteractableStep.java]: Fixed

**Review by Lyfts — APPROVED** [review]

**Pull request merged**